docs/hazmat/primitives/asymmetric/ed25519.rst

```diff
@@ -0,0 +1,76 @@
+.. hazmat::
+
+Ed25519 signing
+===============
+
+.. currentmodule:: cryptography.hazmat.primitives.asymmetric.ed25519
+
+
+Ed25519 is a signing algorithm using `EdDSA`_ and `Curve25519`_. If you do not
+have legacy interoperability concerns then you should strongly consider using
+this signature algorithm.
+
+
+Signing & Verification
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. doctest::
+
+    >>> from cryptography.hazmat.backends import default_backend
+    >>> from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
+    >>> private_key = Ed25519PrivateKey.generate()
+    >>> signature = private_key.sign(b"my authenticated message")
+    >>> public_key = private_key.public_key()
+    >>> # Raises InvalidSignature if verification fails
+    >>> public_key.verify(signature, b"my authenticated message")
+
+Key interfaces
+~~~~~~~~~~~~~~
+
+.. class:: Ed25519PrivateKey
```

Comment:
No serialization?

Reply:
I originally implemented this identically to x25519 for simplicity. We could potentially add serialization here immediately, or add it in a followup PR.